tox.ini

```diff
@@ -6,6 +6,7 @@ usedevelop = true
 deps =
     -rrequirements.txt
     pyftpdlib
+    parameterized
```

Comment:
why do we need an extra dependency here? pytest has paremetrized tests already...

Reply:
The `parametrized` from `pytest` can't be used on class based tests, or at least I wasn't able to use it or find any solution.

Reply:
hmmm... i don't have experience with this myself, but it looks like it's supposed to work according to [docs](https://docs.pytest.org/en/latest/example/parametrize.html#a-quick-port-of-testscenarios). also, the parametrized module docs say:

> Why not use @pytest.mark.parametrize? Because spelling is difficult. Also, parameterized doesn't require you to repeat argument names, and (using param) it supports optional keyword arguments.

"spelling is hard" is not really a good argument in my opinion... ;)

Reply:
I don't get it, should I implement the `pytest_generate_tests`? I wasn't able to figure out how is that supposed to work after looking into the docs for quite a long time. And I think, it affect all methods on the class.
I think, that such implementation would very negatively affect code readability.

The `@pytest.mark.parametrize` decorator doesn't work on class methods.

And the dependency is only testing dependency. It shloudn't be part of the distribution package, so I think it is no such a big deal.

Reply:
@anarcat: [py.test documentation](https://docs.pytest.org/en/latest/unittest.html#pytest-features-in-unittest-testcase-subclasses) states:
> The following pytest features **do not** work, and probably never will due to different design philosophies:
> ...
> - Parametrization

Reply:
Personally, I'd rather see an extra test-only dependency than a mysterious pytest generation function.

Reply:
ah. ouch. well nevermind then, i retract my objection. :)